dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -587,4 +501,59 @@ private static List<OrganisationUnit> getAccessibleDescendants(
   private static boolean isProgramAccessRestricted(Program program) {
     return program != null && (program.isClosed() || program.isProtected());
   }
+
+  private void mapDataElementFilters(EventSearchParams params, List<QueryItem> dataElementFilters) {
+    for (QueryItem item : dataElementFilters) {
+      for (QueryFilter filter : item.getFilters()) {
+        params.addDataElementFilter((DataElement) item.getItem(), filter);
+      }
+    }
+  }
+
+  private void mapAttributeFilters(EventSearchParams params, List<QueryItem> attributeFilters) {
+    for (QueryItem item : attributeFilters) {
+      if (item.getFilters().isEmpty()) {
+        params.addAttribute((TrackedEntityAttribute) item.getItem());
+      }
+
+      for (QueryFilter filter : item.getFilters()) {
+        params.addAttributeFilter((TrackedEntityAttribute) item.getItem(), filter);
+      }
+    }
+  }
+
+  private void mapOrderParam(EventSearchParams params, List<Order> orders)
+      throws BadRequestException {
+    if (orders == null || orders.isEmpty()) {
+      return;
+    }
+
+    for (Order order : orders) {
+      if (order.getField() instanceof String field) {
+        params.orderBy(field, order.getDirection());
+      } else if (order.getField() instanceof UID uid) {
+        DataElement de = dataElementService.getDataElement(uid.getValue());
+        if (de != null) {
+          params.orderBy(de, order.getDirection());
+          continue;
+        }
+
+        TrackedEntityAttribute tea =
+            trackedEntityAttributeService.getTrackedEntityAttribute(uid.getValue());
+        if (tea == null) {
+          throw new BadRequestException(
+              "Cannot order by '"
```

Comment:
Do you think should provide here as well the list of supported fields for ordering?
At this point we cannot really assume that they were trying to ordering on a UID, so the error message could be the same of the one that we throw from the `RequestParamsMapper`.

It could be really strange that they add or remove a char from a `field` and they get completely different error messages

Reply:
I would like to but there are two issues that make it hard
1. these validations are in the (service) mapper. Here we do not have access to the supported field names. The service and store both have access. We cannot add a dependency on the event service here as this would cause a cycle. Adding a dependency on the store here feels odd. Adding direct access to the JdbcEventStore implementation specific orderable fields does not feel right as well.
2. the error message at this point would still be different than the error message from the web layer. This is because we have already translated field names known to users (web view) to the ones that are known to the store (dhis-api).

Reply:
The case you are describing can only happen if the field name they pass in `order` is also a valid UID. I am not sure how often that could happen. I could try to solve this puzzle 😂 

We have documented all the orderable fields in OpenAPI. So in that particular case the answer can be found in the docs.